appscale/tools/local_state.py

```diff
@@ -420,6 +420,29 @@ def update_local_metadata(cls, options, db_master, head_node):
 
 
   @classmethod
+  def clean_local_metadata(cls, keyname):
+    try:
+      with open(cls.get_locations_json_location(keyname), 'r') as file_handle:
+        file_contents = yaml.safe_load(file_handle.read())
```

Comment:
@cdonati yaml.safe_load just does a better job of decoding JSON into python objects, primary example is a JSON load will load strings as unicode strings and string comparisons will fail.